### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by RussKie**
As identified in the last review

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogButtonCollection.cs [hunk 1]
```diff
@@ -46,8 +46,22 @@ public TaskDialogButton Add(string? text, bool enabled = true, bool allowCloseDi
...
         /// <inheritdoc/>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="item"/> is <see langword="null"/>
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="item"/> has already been added to the collection.
+        /// </exception>
+        /// <exception cref="InvalidOperationException">
+        ///   <paramref name="item"/> is already a part of a different collection.
+        /// </exception>
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogButtonCollection.cs]: @BillWagner @gewarren will this work, or do we need to import the original docs?
Here we're subclassing https://docs.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.collection-1.add?view=netcore-3.1

Comment (gewarren) [reply src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogButtonCollection.cs]: Sorry, I don't know the answer. Can we ask @mairaw?

File: src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogButtonCollection.cs [hunk 2]
```diff
@@ -46,8 +46,22 @@ public TaskDialogButton Add(string? text, bool enabled = true, bool allowCloseDi
         }
 
         /// <inheritdoc/>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="item"/> is <see langword="null"/>
```
Comment (gewarren) [inline src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogButtonCollection.cs]:
```suggestion
        ///   <paramref name="item"/> is <see langword="null"/>.
```

File: src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogButtonCollection.cs [hunk 3]
```diff
@@ -73,8 +87,22 @@ protected override void SetItem(int index, TaskDialogButton item)
...
```
Comment (gewarren) [inline src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogButtonCollection.cs]:
```suggestion
        ///   <paramref name="item"/> is <see langword="null"/>.
```

File: src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogProgressBar.cs [hunk 4]
```diff
@@ -49,21 +49,24 @@ public TaskDialogProgressBar(TaskDialogProgressBarState state)
...
+        ///   </para>
+        ///   <para>
+        ///     This property can be set while the dialog is shown. However, while the dialog is
+        ///     shown, it is not possible to change the state from
+        ///     <see cref="TaskDialogProgressBarState.None"/> to any other state,
+        ///     and vice versa.
+        ///   </para>
         /// </remarks>
+        /// <exception cref="InvalidOperationException">
+        ///   Attempt to hide the progress bar by setting the value
```
Comment (gewarren) [inline src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogProgressBar.cs]: This doesn't read quite right to me. Maybe:

```suggestion
        ///   The value is <c>TaskDialogProgressBarState.None</c> and the dialog is displayed.
```

File: src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogRadioButtonCollection.cs [hunk 5]
```diff
@@ -45,8 +45,22 @@ public TaskDialogRadioButton Add(string? text)
         }
 
         /// <inheritdoc/>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="item"/> is <see langword="null"/>
```
Comment (gewarren) [inline src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogRadioButtonCollection.cs]:
```suggestion
        ///   <paramref name="item"/> is <see langword="null"/>.
```

File: src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogRadioButtonCollection.cs [hunk 6]
```diff
@@ -72,8 +86,22 @@ protected override void SetItem(int index, TaskDialogRadioButton item)
...
```
Comment (gewarren) [inline src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogRadioButtonCollection.cs]:
```suggestion
        ///   <paramref name="item"/> is <see langword="null"/>.
```

File: src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogProgressBar.cs (hunk 4 above)
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogProgressBar.cs]: Updated, thank you

File: src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogButtonCollection.cs (hunk 1 above)
Comment (mairaw) [reply src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogButtonCollection.cs]: inheritdoc still doesn’t work on mdoc. You could also keep here as-is and copy the docs on the API repo.

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogButtonCollection.cs]: Thank you.

@kpreisser we should be good to go.

**Comment by kpreisser** [comment]
Thanks a lot!

**Pull request merged**